src/resmom/requests.c

```diff
@@ -3392,9 +3392,24 @@ req_cpyfile(struct batch_request *preq)
 		}
 	} else {
 		if (stage_inout.bad_files) {
-			sprintf(log_buffer, "Job files not copied: %s", stage_inout.bad_list);
+			char *token = NULL; 
+			char *rest = stage_inout.bad_list; 
+			char *save_ptr = NULL;
 			log_event(PBSEVENT_DEBUG2, PBS_EVENTCLASS_JOB, LOG_DEBUG,
-					dup_rqcpf_jobid, log_buffer);
+				dup_rqcpf_jobid, "Job files not copied:---->>>>");
+			token = strtok_r(rest, "\n", &save_ptr);
+			while (token != NULL) {
+				char *temp_buff = NULL;
+				if ((pbs_asprintf(&temp_buff, "%s\n", token)) != -1) {
```

Comment:
If pbs_asprintf() returns -1, you don't advance token.  pbs_asprintf() will likely continue to return -1, and this will become an infinite loop.  You should move the strtok_r() outside of the if statement, break out of the loop, or bail out of the function if pbs_asprintf() returns -1.  It does that on a malloc() failure.